### PR TITLE
Force test failure on unhandled instrumented exception

### PR DESCRIPTION
**Pull request by amarziali**
# What Does This Do

# Motivation

# Additional Notes

Jira ticket: [PROJ-IDENT]

### CODE REVIEW

**Review by richardstartin — APPROVED** [review]

**Review by mcculls — APPROVED** [review]
Nice! Especially like the idea to count these errors,  as we can then expose that as a metric :)

File: dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/ExceptionHandlers.java [hunk 1]
```diff
@@ -107,8 +117,22 @@ public Size apply(final MethodVisitor mv, final Implementation.Context context)
               return size;
             }
           });
+    }
+
+    public AtomicLong getErrorCounter() {
```
Comment (mcculls) [inline dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/ExceptionHandlers.java:122]: You could move the counter to a static field in `ExceptionHandlers` and expose static methods to increment it and get the current count. That would avoid the need for `DefaultExceptionHandler`

Comment (amarziali) [reply dd-java-agent/agent-tooling/src/main/java/datadog/trace/agent/tooling/bytebuddy/ExceptionHandlers.java]: I was sticking to the instance we return as singleton but i's not really important to respect this. I applied the suggestion I will simplify things thanks

**Review by mcculls — APPROVED** [review]

**Comment by amarziali** [comment]
@smola as discussed there are ASM tests failing because of "Interaction with TraceSegment after root span has already finished"

Till today this error was triggered but the exception handler did not record the failure. Today is happening and the tests are failing. This check makes sense when strict trace writing is enabled. Now, I aligned it and disabled strict trace writing for grizzly tests that are showing those failures. A fixme is reminding that at long term we should probably fix that instrumentation

**Review by smola — APPROVED** [review]

**Pull request merged**